libs/shared/ui/src/lib/components/buttons/button-menu/button-menu.tsx

```diff
@@ -0,0 +1,114 @@
+import { IconEnum } from '@console/shared/enums'
+import { Icon } from '@console/shared/ui'
+import { useState } from 'react'
+import { Link } from 'react-router-dom'
+import Menu, { MenuAlign } from '../../menu/menu'
+import { MenuItemProps } from '../../menu/menu-item/menu-item'
+
+export enum ButtonMenuStyle {
+  BASIC = 'basic',
+  RAISED = 'raised',
+  STROKED = 'stroked',
+  FLAT = 'flat',
+}
+
+export interface ButtonMenuProps {
+  children: React.ReactNode
+  style?: ButtonMenuStyle
+  iconRight?: IconEnum | string
+  link?: string
+  disabled?: boolean
+  className?: string
+  onClick?: () => void
+  external?: boolean
+  loading?: boolean
+  menus: { items: MenuItemProps[]; title?: string; button?: string; buttonLink?: string; search?: boolean }[]
+}
+
+export function ButtonMenu(props: ButtonMenuProps) {
```

Comment:
Some improvements: 
- Need to remove loading (useless here)
- We don't have an external link for this button
- Style not good, missing hover transition, and the "plus" don't have the good size